bezier-rs/lib/src/lib.rs

```diff
@@ -624,37 +641,41 @@ impl Bezier {
 
 	/// Implementation of the algorithm to find curve intersections by iterating on bounding boxes.
 	/// - `self_original_t_interval` - Used to identify the `t` values of the original parent of `self` that the current iteration is representing.
-	/// Note that the `t` interval the other curve is not needed since we want to return `t` with respect to it.
-	fn intersections_between_subcurves(&self, self_original_t_interval: [f64; 2], other: &Bezier, error: f64) -> Vec<f64> {
+	/// - `other_original_t_interval` - Used to identify the `t` values of the original parent of `other` that the current iteration is representing.
+	fn intersections_between_subcurves(&self, self_original_t_interval: Range<f64>, other: &Bezier, other_original_t_interval: Range<f64>, error: f64) -> Vec<[f64; 2]> {
 		let bounding_box1 = self.bounding_box();
 		let bounding_box2 = other.bounding_box();
 
 		// Get the `t` interval of the original parent of `self` and determine the middle `t` value
-		let [curve1_start_t, curve1_end_t] = self_original_t_interval;
-		let curve1_mid_t = curve1_start_t + (curve1_end_t - curve1_start_t) / 2.;
+		let Range { start: self_start_t, end: self_end_t } = self_original_t_interval;
+		let self_mid_t = self_start_t + (self_end_t - self_start_t) / 2.;
+
+		// Get the `t` interval of the original parent of `other` and determine the middle `t` value
+		let Range {
+			start: other_start_t,
+			end: other_end_t,
+		} = other_original_t_interval;
+		let other_mid_t = other_start_t + (other_end_t - other_start_t) / 2.;
```

Comment:
(both this one and the one a few lines above)
```rs
let other_mid_t = (other_end_t + other_start_t) / 2.;
```
is equivalent, and more efficient:
![image](https://user-images.githubusercontent.com/4388688/182052055-4ea0d240-b131-4d19-aa82-d6263021b16b.png)